File: src/test/java/io/tripled/social/client/presentation/controller/ReadControllerTest.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.application.SpiedReadMessagesUseCase;
import io.tripled.social.client.presentation.TestOutput;
import io.tripled.social.client.presentation.request.ReadCliRequest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;

class ReadControllerTest {

  private ReadController readController;
  private SpiedReadMessagesUseCase service;

  @BeforeEach
  void setUp() {
    service = new SpiedReadMessagesUseCase();
    readController = new ReadController(service);
  }

  @Test
  void executesCommand() {
    ReadCliRequest request = new ReadCliRequest("Alice");

    readController.execute(request, new TestOutput());

    assertThat(service.userName, equalTo("Alice"));
  }

}

File: src/test/kotlin/io/tripled/social/client/presentation/controller/ReadControllerTest.kt
package io.tripled.social.client.presentation.controller

import io.tripled.social.client.application.SpiedReadMessagesUseCase
import io.tripled.social.client.presentation.TestOutput
import io.tripled.social.client.presentation.request.ReadCliRequest
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class ReadControllerTest {
    private var readController: ReadController? = null
    private var service: SpiedReadMessagesUseCase? = null

    @BeforeEach
    fun setUp() {
        service = SpiedReadMessagesUseCase()
        readController = ReadController(service!!)
    }

    @Test
    fun executesCommand() {
        val request = ReadCliRequest("Alice")

        readController!!.execute(request, TestOutput())

        MatcherAssert.assertThat<String?>(service!!.userName, Matchers.equalTo<String?>("Alice"))
    }
}